File: api/Core/Model/Permission.cs
using System;

namespace Xcianify.Core.Model
{
    public class Permission : BaseModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ModuleName { get; set; }
        public string ActionType { get; set; }
    }
}

File: pharma-erp-3/validations/permission.ts
import { z } from "zod"

export const permissionFormSchema = z.object({
  name: z.string().min(1, { message: "Permission name is required" }),
  description: z.string().optional(),
  moduleName: z.string().min(1, { message: "Module name is required" }),
  actionType: z.preprocess(
    (val) => Array.isArray(val) ? val.join(",") : val,
    z.string().min(1, { message: "Action type is required" })
  ),
  isActive: z.boolean()
})

export type PermissionFormValues = z.infer<typeof permissionFormSchema>;
